Assets/Scripts/Dialogue/InteractiveNPC.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class InteractiveNPC : MonoBehaviour
+{
+    private bool IsDisplayingDialogue;
+    private bool IsPlayerInRange;
+    private float DistanceBetweenPlayer;
+    public float MinDistance = 3.5f;
+    public DialogueManager DialogueManager;
+    public Conversation Conversation;
+
+    void Start()
+    {
+        IsDisplayingDialogue = false;
+        IsPlayerInRange = false;
+        DistanceBetweenPlayer = 0f;
+    }
+
+    void Update()
+    {
+        CheckIfInRange();
+
+        if(IsDisplayingDialogue & Input.GetButtonDown("Action Command"))
+        {
+            IsDisplayingDialogue = DialogueManager.AdvanceConversation();
+        }
+        else if (Input.GetButtonDown("Action Command") && IsPlayerInRange)
+        {
+            IsDisplayingDialogue = !IsDisplayingDialogue;
+            DialogueManager.StartDialogue(Conversation);
+        }
+    }
+
+    void CheckIfInRange()
+    {
+        GameObject player = GameObject.FindWithTag("Player");
```

Comment:
This player ref can be fetched on `Start` once instead of on each method call. Also, when we get that 2nd player, this is not scalable yet. Fetch the list of players maybe?